### PR TITLE
Fix header.hash for height 1

### LINKED ISSUE #436: light-node: can't initialise with header at height 1
An attempt to init with the header at height `1` fails.

Potentially reproducible with:
* run a tendermint full node
* `curl -X GET "http://localhost:26657/block?height=1” -H  "accept: application/json" `
* deserialise in Rust and investigate why `header.hash() != block_id.hash`

Potential causes:
* latest 0.33 changed something
* a potential bug when one of the `last_` fields is not present in

https://github.com/informalsystems/tendermint-rs/blob/142796b0cc37319f91af55cf2703b96a0cb80230/tendermint/src/lite_impl/header.rs#L28-L58

### PR DESCRIPTION
**Pull request by liamsi**
closes #436 

* [x] Referenced an issue explaining the need for the change
* [ ] Updated all relevant documentation in docs
* [ ] Updated all code comments where relevant
* [x] Wrote tests
* [ ] Updated CHANGES.md

### CODE REVIEW

File: rpc/tests/integration.rs [hunk 1]
```diff
@@ -224,6 +224,16 @@ mod endpoints {
         assert_eq!(header.hash(), block_id.hash);
     }
 
+    #[test]
+    fn commit_height_1() {
```
Comment (xla) [inline rpc/tests/integration.rs:228]: Nice to have a case, when I tried it out I went straight into the `lite_impl` module. As we likely gonna fix the `fn hash` implementation, shouldn't we test deeper in the stack?

Comment (liamsi) [reply rpc/tests/integration.rs:228]: Probably both?

File: tendermint/src/lite_impl/header.rs [hunk 2]
```diff
@@ -48,28 +48,80 @@ impl lite::Header for block::Header {
...
-        fields_bytes.push(bytes_enc(self.proposer_address.as_bytes()));
+        fields_bytes.push(encode_bytes(self.proposer_address.as_bytes()));
 
         Hash::Sha256(simple_hash_from_byte_vectors(fields_bytes))
     }
 }
 
-fn bytes_enc(bytes: &[u8]) -> Vec<u8> {
-    let mut chain_id_enc = vec![];
-    prost_amino::encode_length_delimiter(bytes.len(), &mut chain_id_enc).unwrap();
```
Comment (liamsi) [inline tendermint/src/lite_impl/header.rs:62]: The problem was here: we length-encoded even zero length bytes. In this particular case the `app_hash`.

**Comment by liamsi** [comment]
Thanks @romac! I Forgot that this was copied over. The deduplication should probably go in the other direction (into the light-client crate) but it makes sense to have only one tested and working implementation in place 👍

**Pull request merged**